cadcutils/cadcutils/util/utils.py

```diff
@@ -316,24 +320,31 @@ def get_base_parser(subparsers=True, version=None, usecert=True,
                             help='name of user to authenticate. ' +
                                  'Note: application prompts for the '
                                  'corresponding password!')
-    cparser.add_argument('--host',
-                         help='base hostname for services - used mainly '
-                              'for testing (default: '
-                              'www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca)',
-                         default='www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca')
-    if default_resource_id is None:
-        cparser.add_argument('--resource-id',
-                             type=urlparse, required=True,
-                             help='resource identifier '
-                                  '(e.g. ivo://cadc.nrc.ca/service)')
+    cparser.add_argument('--host', help=SUPPRESS)
+    cparser.add_argument('-k', '--insecure', action='store_true',
+                         help=SUPPRESS)
+    if service is None:
+        if default_resource_id is None:
+            cparser.add_argument('--resource-id',
+                                 type=urlparse, required=True,
+                                 help='resource identifier '
+                                      '(e.g. ivo://cadc.nrc.ca/service)')
+        else:
+            cparser.add_argument('--resource-id', type=parse_resource_id,
+                                 default=default_resource_id,
+                                 help='resource identifier (default {})'.
+                                 format(default_resource_id))
     else:
-        cparser.add_argument('--resource-id', type=parse_resource_id,
-                             default=default_resource_id,
-                             help='resource identifier (default {})'.format(
-                                 default_resource_id))
+        cparser.add_argument(
```

Comment:
Would it be helpful to a user if the help text included a URL where the list of available services could be found? Does such a URL exist?

Reply:
The library already displays a list of valid service IDs when the one specified by the user is not found. It would be nice if it could display only the relevant ones (`VOSpace` for `vos`, `TAP` for `cadc tap` etc.). That was a feature we've discussed in the past but we god nowhere yet.